pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -220,9 +220,7 @@ self: super: {
   digit = doJailbreak super.digit;
 
   # 2020-06-05: HACK: does not passes own build suite - `dontCheck`
-  hnix = generateOptparseApplicativeCompletion "hnix" (
-    dontCheck super.hnix
-    );
```

Comment:
Do you know why `generateOptparseApplicativeCompletion` doesn't work for `hnix`?  Is this a problem with `generateOptparseApplicativeCompletion`, or something that needs to be fixed upstream in `hnix`?

If it is the former, can you file an issue about `generateOptparseApplicativeCompletion`?  If it is the later, can you file an issue with `hnix`?

Other than that, this looks good to me.

Reply:
Now that you ask about it, I realize what the problem is.
hnix does not produce an executable anymore. I have no clue why. But that certainly explains why it can't have completion...

Reply:
Huh, that's somewhat strange.  Maybe there was some change upstream?

I'm happy to merge this in then, if `hnix` is no longer expected to contain an executable.

Reply:
I will dig deeper maybe this is indicative of a larger issue. So let's not just fix the symptom.

Reply:
Okay I found the culprit in the executable definition.:
```
  if impl(ghc < 8.10)
    -- GHC < 8.10 comes with haskeline < 0.8, which we don't support.
    -- To simplify CI, we just disable the component.
    buildable: False
```

So what we actually want is an exception for ghc < 8.10 to don‘t build the completion.

Reply:
Ah, in that case you can add an override in `pkgs/development/haskell-modules/configuration-ghc-8.10.x.nix` that will only apply for ghc810.

Reply:
That does not sound very future proof. likely we will forget to transfer it to 8.12.

Reply:
I have done it this way anyway, because I couldn‘t think of anything better.

Reply:
Do not worry about it, team would always remember to keep `hnix` properly in Nixpkgs, since it is related project. I would add autocompletions in time, when Nixpkgs switches to newer GHCs. We also plan to provide executable in its own package in the future, when it in fact would work and when typing interfaces settle down.